### PR TITLE
07779 Replace codecs with protobuf

### LINKED ISSUE #7779: Replace LongCodec, StringCodec, BytesCodec, and ListCodec with protobuf
General design principle for modular services is to use Protobuf for all state values.  We have a few uses of Bytes, String, Long, and one List that need to be stored in state.  Initially this was done with Codec implementations for expedience, but we need to replace those with protobuf definitions.

Suggested structure
```
services/state
    |-- primitives.proto
        %-- ProtoLong
        %-- ProtoInteger
        %-- ProtoBoolean
        %-- ProtoString
        %-- ProtoBytes
    |-- schedule
        |-- schedule.proto
            %-- ScheduleList
```

Once the protobuf changes are merged to the `add-pbj-types-for-state` branch and complete, then modify the services codebase to remove the Codec implementations and replace with the protobuf implementations.

### PR DESCRIPTION
**Pull request by jsync-swirlds**
* Replace the LongCodec, StringCodec, BytesCodec, and ListCodec classes recently added with protobuf alternatives.
   * LongCodec -> ProtoLong
   * StringCodec -> ProtoString
   * BytesCodec -> ProtoBytes
   * ListCodec -> ScheduleList (this was only used by schedule service).
 * Removed the tests for the codec classes
   * The LongCodec and StringCodec were moved to TestLongCodec and TestStringCodec in app tests because the App module MerkleTestBase is written to depend on Codecs and that spider through a ton of other tests.
 * Updated module-info in app-spi to remove the export for the now empty codec package
 * Fixed code and tests in File service to use ProtoBytes instead of Bytes and BytesCodec
 * Fixed code and tests in NetworkAdmin service to use ProtoBytes instead of Bytes and BytesCodec
 * Fixed code and tests Schedule service to use ProtoString, ProtoLong, and ScheduleList instead of StringCodec, LongCodec, and ListCodec.
 * Fixed use of StringCodec in TokenServiceImpl to use ProtoString.
   * Note: Token service still used a codec EntityNumCodec but that was switched to EntityNumber separately.

Resolves #7779

Note:  The protobuf changes were merged last week.

### CODE REVIEW

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/merkle/TestStringCodec.java
Comment (kimbor) [inline hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/merkle/TestStringCodec.java:1]: This is a test fixture class, right? I think it should be located in com.hedera.node.app.fixtures?

Comment (jsync-swirlds) [reply hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/merkle/TestStringCodec.java:1]: All I'm doing here for TestLongCodec and TestStringCodec is fixing the pre-existing use of a Codec class that was not correct and should be removed, but is more work than we can afford at the moment.
I'd rather not encourage additional use of this codec by promoting it to a fixture.

Comment (kimbor) [reply hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/merkle/TestStringCodec.java:1]: Ok, then maybe add a comment to that affect at the top of the class. I found it confusing that this class was located here. Same for the other Test*Codec classes.

Comment (kimbor) [reply hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/merkle/TestStringCodec.java:1]: Or move them to the fixtures package and mark them @deprecated

Comment (jsync-swirlds) [reply hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/merkle/TestStringCodec.java:1]: Adding deprecated produces a ton of warnings that just add more noise to the thousands of lines we already get.  I'll add some comments regarding use and what needs to be done when time permits.

Comment (jsync-swirlds) [reply hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/merkle/TestStringCodec.java:1]: Done.

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/merkle/TestLongCodec.java [hunk 1]
```diff
@@ -23,11 +23,19 @@
...
+ * <strong>This class should not be used elsewhere.</strong>
+ * The use of Codec implementations is not ideal, and we have the protobuf ProtoLong as
+ * a replacement.  When time permits this class needs to be removed and all usages replaced
+ * with ProtoLong and the ProtoLong.PROTOBUF codec.
+ * @deprecated Use ProtoLong and ProtoLong.PROTOBUF instead of Long and this codec.
+ */
 @SuppressWarnings("Singleton")
-public class LongCodec implements Codec<Long> {
-    public static final LongCodec SINGLETON = new LongCodec();
+public class TestLongCodec implements Codec<Long> {
```
Comment (netopyr) [inline hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/merkle/TestLongCodec.java:35]: Is it possible to make these classes package private to enforce their limited use?

**Review by netopyr — DISMISSED** [review]
LGTM in regards to `hedera-base`

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/merkle/TestLongCodec.java [hunk 2]
```diff
@@ -23,11 +23,19 @@
...
 import java.util.Objects;
 
+/**
+ * This is a test-only class used exclusively by MerkleTestBase.
+ * <strong>This class should not be used elsewhere.</strong>
+ * The use of Codec implementations is not ideal, and we have the protobuf ProtoLong as
+ * a replacement.  When time permits this class needs to be removed and all usages replaced
+ * with ProtoLong and the ProtoLong.PROTOBUF codec.
+ * @deprecated Use ProtoLong and ProtoLong.PROTOBUF instead of Long and this codec.
+ */
```
Comment (mhess-swl) [inline hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/merkle/TestLongCodec.java:33]: thank you for clarifying why this class exists!

File: hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/ScheduleServiceImpl.java [hunk 3]
```diff
@@ -54,40 +53,45 @@ public void registerSchemas(@NonNull final SchemaRegistry registry) {
...
+        }
 
-    private StateDefinition<Long, List<Schedule>> schedulesByExpirySec() {
-        return StateDefinition.inMemory(SCHEDULES_BY_EXPIRY_SEC_KEY, LongCodec.SINGLETON, LIST_OF_SCHEDULE_CODEC);
-    }
+        @SuppressWarnings("rawtypes")
+        @NonNull
+        @Override
+        public Set<StateDefinition> statesToCreate() {
+            final Codec<MerkleScheduledTransactionsState> migrationCodec = MonoMapCodecAdapter.codecForSelfSerializable(
```
Comment (mhess-swl) [inline hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/ScheduleServiceImpl.java:68]: Why do we still need this mono migration codec?

Comment (jsync-swirlds) [reply hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/ScheduleServiceImpl.java:68]: I don't think we do, but it's not the subject of this PR.
I believe I have it removed in the next PR for schedule service.

File: hedera-node/hedera-file-service-impl/src/test/java/com/hedera/node/app/service/file/impl/test/handlers/FileUpdateTest.java [hunk 4]
```diff
@@ -351,8 +352,10 @@ void appliesNewContentUpgradeFile() {
         subject.handle(handleContext);
 
         final var newFile = writableUpgradeStates.poll();
-        assertEquals(newContent.length(), newFile.length());
-        assertEquals(newContent.toString(), newFile.toString());
+        assertNotNull(newFile);
+        assertNotNull(newFile.value());
+        assertEquals(newContent.length(), newFile.value().length());
+        assertEquals(newContent.toString(), newFile.value().toString());
```
[the remark this answers is not in the document]
Comment (jsync-swirlds) [reply hedera-node/hedera-file-service-impl/src/test/java/com/hedera/node/app/service/file/impl/test/handlers/FileUpdateTest.java:358]: I'm deliberately avoiding any semantic change to the other services (File, Network Admin, Token) as I don't have the full understanding to be sure I don't break something.
This PR shouldn't be touching anything not strictly required to remove the Codec implementations used for primitives.

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/merkle/TestStringCodec.java
Comment (kimbor) [reply hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/merkle/TestStringCodec.java:1]: Looks good

**Review by kimbor — APPROVED** [review]

**Review by povolev15 — APPROVED** [review]
LGTM

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/merkle/TestLongCodec.java (hunk 2 above)
Comment (jsync-swirlds) [reply hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/merkle/TestLongCodec.java:33]: Thank @kimbor who suggested it 😁

**Review by mhess-swl — DISMISSED** [review]

**Review by mhess-swl — APPROVED** [review]

**Review by kimbor — APPROVED** [review]

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM ! Regarding the comment in PR description, for using `EntityNumCodec` in TokenService, it should have been deleted in my PR that just got merged.

**Comment by jsync-swirlds** [comment]
> LGTM ! Regarding the comment in PR description, for using `EntityNumCodec` in TokenService, it should have been deleted in my PR that just got merged.

Yes, that was the conflict in the most recent rebase 😄

**Review by rbair23 — APPROVED** [review]

**Pull request merged**